This patch has been reverted later on by
http://git.denx.de/?p=u-boot.git;a=commit;h=e3c52f2b8779469c843eb79282396f1a5ca3fef5

but we still need it to build the tools without having to configure
a full u-boot environment.

Signed-off-by: Markos Chandras <hwoarang@gentoo.org>

From: Mike Frysinger <vapier@gentoo.org>
Date: Sat, 10 Nov 2012 19:47:47 +0000 (+0000)
Subject: fw_env: fix building w/out a config.h
X-Git-Tag: v2013.01-rc3~19
X-Git-Url: http://git.denx.de/?p=u-boot.git;a=commitdiff_plain;h=92ace272d06cec1d7f1533bb9edf914fb5845fba;hp=7a546db2ffc6a713fb8009d8246c29da5420f6a7

fw_env: fix building w/out a config.h

Signed-off-by: Mike Frysinger <vapier@gentoo.org>
Acked-by: Peter Korsgaard <jacmet@sunsite.dk>
---

diff --git a/tools/env/Makefile b/tools/env/Makefile
index 0e798e0..e6dc5f3 100644
--- a/tools/env/Makefile
+++ b/tools/env/Makefile
@@ -26,7 +26,7 @@ include $(TOPDIR)/config.mk
 HOSTSRCS := $(SRCTREE)/lib/crc32.c  fw_env.c  fw_env_main.c
 HOSTSRCS += $(SRCTREE)/lib/ctype.c $(SRCTREE)/lib/linux_string.c
 HOSTSRCS += $(SRCTREE)/common/env_attr.c $(SRCTREE)/common/env_flags.c
-HEADERS	:= fw_env.h $(OBJTREE)/include/config.h
+HEADERS	:= fw_env.h
 
 # Compile for a hosted environment on the target
 HOSTCPPFLAGS  = -idirafter $(SRCTREE)/include \
@@ -36,6 +36,15 @@ HOSTCPPFLAGS  = -idirafter $(SRCTREE)/include \
 		-DUSE_HOSTCC \
 		-DTEXT_BASE=$(TEXT_BASE)
 
+# Pass CONFIG_xxx settings via the command line so that we can build w/out
+# a config.h file existing in the first place.  Useful for generic builds.
+CONFIG_VARS_TO_PASS = \
+	ENV_OVERWRITE \
+	OVERWRITE_ETHADDR_ONCE \
+	ETHADDR
+HOSTCPPFLAGS += \
+	$(foreach x,$(CONFIG_VARS_TO_PASS),$(if $(CONFIG_$(x)),-DCONFIG_$(x)=$(CONFIG_$(x))))
+
 ifeq ($(MTD_VERSION),old)
 HOSTCPPFLAGS += -DMTD_OLD
 endif
diff --git a/tools/env/fw_env.h b/tools/env/fw_env.h
index a1a6807..19703c7 100644
--- a/tools/env/fw_env.h
+++ b/tools/env/fw_env.h
@@ -21,15 +21,6 @@
  * MA 02111-1307 USA
  */
 
-/* Pull in the current config to define the default environment */
-#ifndef __ASSEMBLY__
-#define __ASSEMBLY__ /* get only #defines from config.h */
-#include <config.h>
-#undef	__ASSEMBLY__
-#else
-#include <config.h>
-#endif
-
 /*
  * To build the utility with the static configuration
  * comment out the next line.
@@ -52,22 +43,6 @@
 #define DEVICE2_ENVSECTORS     2
 #endif
 
-#ifndef CONFIG_BAUDRATE
-#define CONFIG_BAUDRATE		115200
-#endif
-
-#ifndef CONFIG_BOOTDELAY
-#define CONFIG_BOOTDELAY	5	/* autoboot after 5 seconds	*/
-#endif
-
-#ifndef CONFIG_BOOTCOMMAND
-#define CONFIG_BOOTCOMMAND							\
-	"bootp; "								\
-	"setenv bootargs root=/dev/nfs nfsroot=${serverip}:${rootpath} "	\
-	"ip=${ipaddr}:${serverip}:${gatewayip}:${netmask}:${hostname}::off; "	\
-	"bootm"
-#endif
-
 extern int   fw_printenv(int argc, char *argv[]);
 extern char *fw_getenv  (char *name);
 extern int fw_setenv  (int argc, char *argv[]);
